hypothesis-python/src/hypothesis/internal/escalation.py

```diff
@@ -67,9 +69,10 @@ def get_trimmed_traceback(exception=None):
     # was raised inside Hypothesis. Additionally, the environment variable
     # HYPOTHESIS_NO_TRACEBACK_TRIM is respected if nonempty, because verbose
     # mode is prohibitively slow when debugging strategy recursion errors.
+    assert hypothesis.settings.default is not None
     if (
         tb is None
-        or os.environ.get("HYPOTHESIS_NO_TRACEBACK_TRIM", None)
+        or os.environ.get("HYPOTHESIS_NO_TRACEBACK_TRIM")
         or hypothesis.settings.default.verbosity >= hypothesis.Verbosity.debug
```

Comment:
something to check: I think `settings.default` might be the _static_ default settings, and for _current_ settings we'd want to use `settings().verbosity`.  If so we should audit all uses of `settings.default`...

Reply:
I think this is ok. `settings.load_profile()` and `with local_settings` both set `settings.default`, and `settings.__init__` uses `parent = parent or settings.default` to inherit settings, so `settings()` and `settings.default` should be equivalent.

That said - `settings()` avoids `assert settings.default is not None` for type checking, so I'd prefer `settings()` anyway.